src/manifest/classes/period.ts

```diff
@@ -163,6 +163,10 @@ export default class Period implements IPeriodMetadata {
           // Go to next adaptation as an optimisation measure.
           // NOTE this only is true if we never change a codec from supported
           // to unsuported and its opposite.
+
+          if (adaptation.supportStatus.hasSupportedCodec === true) {
+            hasSupportedAdaptations = true;
+          }
```

Comment:
Shouldn't we also handle the `undefined` case here when it was `false` before technically?

Even if it changes nothing in the current case.

Reply:
I think it's already handled, because if `hasSupportedCodec` is `undefined`, then `hasCodecWithUndefinedSupport` should be `true` and we are not in this condition

Reply:
I guess you're right